### ISSUE TITLE
Rails/HasManyOrHasOneDependent false positive when passing a scope

### ISSUE THREAD

**Issue by francois-ferrandis** [rubocop/rubocop#10073]
## Expected behavior

No cop detected

## Actual behavior

`Rails/HasManyOrHasOneDependent` is detected

## Steps to reproduce the problem

Given this model:

```ruby
class Message < ApplicationRecord
  has_many :comments, -> { order(:created_at) }, dependent: :destroy, inverse_of: :message
end
```

I get:

```
app/models/message.rb:2:3: C: Rails/HasManyOrHasOneDependent: Specify a :dependent option.
  has_many :comments, -> { order(:created_at) }, dependent: :destroy, inverse_of: :message
  ^^^^^^^^
```

## RuboCop version

```
1.20.0 (using Parser 3.0.2.0, rubocop-ast 1.11.0, running on ruby 3.0.2 x86_64-linux)
  - rubocop-rails 2.12.1
```
